Patch 1

Bluscream opened the pull request:
- New setting to disable relays in the config
- Some logging at the start to have a quick overview of what gets redirected where
- New setting to bind the relay to a specific address.

![https://i.imgur.com/RxzesK2.png](https://i.imgur.com/RxzesK2.png)

Bluscream committed:
Update config.yml

Bluscream committed:
Update tcp.py

Bluscream committed:
Update udp.py

derkalle4 commented:
Hey @Bluscream,

thanks for your PR. You've missed the ts3proxy.py otherwise the classes will not work like expected. And If you rename "relayAddr" to "relayAddress" please also rename them in the source code. I personally would prefer "relayAddr" because its a little bit shorter and you also get the meaning of it.
At last I would enable all relays by default instead of disabling the serverquery. Every user could switch them off in the config.

Bluscream committed:
Update ts3proxy.py

Bluscream committed:
renamed remoteAdd to remoteAddress in the code too

Bluscream committed:
renamed remoteAdd to remoteAddress in the code too

Bluscream committed:
Enabled ServerQuery proxy by default

I recommend to use a SSH proxy (like [YaTQA](http://yat.qa/features/#Overview_of_Features_That_Require_a_Key) for example tho since telnet is pretty insecure)

Bluscream commented:
Hey @derkalle4,

Ich hab mich um deine Vorschläge gekümmert :)
BTW: Address macht für Leute mit keiner bis wenig ahnung durchaus Sinn ^^

Bluscream committed:
Create setup.sh

Bluscream committed:
Added code tags to readme

+ use setup.py

Bluscream committed:
Renamed UDP to Voice

derkalle4 commented:
Hey @Bluscream,

mehrere Verbesserungsvorschläge (dafür auch ein Dank an @AlexElvers der ein wachendes Auge über mich hält hehe):

- Auch wenn ich es selber (leider noch) nicht ganz einhalte sollte der Source Code Style PEP8  (https://www.python.org/dev/peps/pep-0008/) entsprechen 
- In ts3proxy.py musst du die Listen für die Werte nicht erst instanzieren und danach befüllen, das kann man direkt tun (ist kürzer):
`[arg1, arg2, arg3]`
- nach PEP8 sollten Variablen lowercase sein (also statt UDP dann udp) und am besten sprechende Namen verwenden sowie statt CamelCase lieber _ (das mache ich da auch noch falsch). Will also heißen: statt UDP = [] lieber sowas wie ts3_udp_args = [] und statt FT = [] lieber file_transfer_args = []

Zur Not übernehme ich dein Pull Request und formatiere das nochmal neu :) soll auch nicht böse gemeint sein, ich lerne ja gerade auch noch so einiges über Python.

Edit: ich werde dann später auch noch deine Debug-Messages direkt in die Klassen auslagern und diese dann in der Config optional machen. So kann jeder selbst entscheiden wie viel Output er haben möchte ;)

AlexElvers commented:
Ergänzung:

Statt
```python
Udp(UDP[0],UDP[1],UDP[2])
```
geht auch
```python
Udp(*UDP[:3])
```
und statt
```python
print("UDP: %s:%s <-> %s:%s"%(UDP[0],UDP[1],UDP[2],UDP[3]))
```
geht auch
```python
print("UDP: %s:%s <-> %s:%s" % UDP)
```

Bluscream commented:
> In ts3proxy.py musst du die Listen für die Werte nicht erst instanzieren und danach befüllen, das kann man direkt tun (ist kürzer):
[arg1, arg2, arg3]

Hab ich probiert hat mir nen syntaxerror ausgespuckt.
Wegen PEP8 bin ich ehrlich gesagt kein fan von. Ich finde es macht code teilweise sogar unleserlicher (Also das würd ich dir überlassen)

Bluscream committed:
Update ts3proxy.py

Bluscream committed:
Update ts3proxy.py
